src/containers/Transaction/Transaction.js

```diff
@@ -0,0 +1,174 @@
+      setValue('total', total.toFixed(2));
+    }
+  }, [amount, currencyName, errors.amount, getValues, setValue, stockExchangeData]);
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} className={styles.wrapper}>
+      <div className={styles.heading}>
+        <Text text="Transaction Type:" type="HEADING_5" />
+        <Text text={isBuying ? 'Buy' : 'Sell'} type="HEADING_5" state={isBuying ? 'SUCCESS' : 'ACCENT'} />
+        <Controller
+          name="isBuying"
+          control={control}
+          render={({ field }) => (
+            <Component.Switch
+              color="primary"
+              checked={field.value}
+              inputProps={{ 'aria-label': 'primary checkbox' }}
+              {...field}
+            />
+          )}
+        />
+      </div>
+      <div className={styles.form}>
+        <Component.FormControl variant="outlined" error={!!errors.currencyName}>
+          <Component.InputLabel id="currencyName">Name</Component.InputLabel>
+          <Controller
+            name="currencyName"
+            control={control}
+            render={({ field }) => (
+              <Component.Select
+                labelId="currencyName"
+                id="currencyName"
+                label="currencyName"
+                {...field}
+              >
+                {stockExchangeData.map((dataItem) => (
+                  <Component.MenuItem key={dataItem.uuid} value={dataItem.name}>
+                    {dataItem.name}
+                  </Component.MenuItem>
+                ))}
+              </Component.Select>
+            )}
+          />
+          <Component.FormHelperText>
+            {errorMessages[errors.currencyName?.type]}
+          </Component.FormHelperText>
+        </Component.FormControl>
+        <Controller
+          name="price"
+          control={control}
+          render={({ field }) => (
+            <Component.TextField
+              id="price"
+              label="Price"
+              variant="outlined"
+              disabled
+              {...field}
+            />
+          )}
+        />
+        <Controller
+          name="amount"
+          control={control}
+          render={({ field }) => (
+            <Component.TextField
+              id="amount"
+              type="number"
+              label="Amount"
+              variant="outlined"
+              error={!!errors.amount}
+              helperText={errorMessages[errors.amount?.type]}
+              {...field}
+            />
+          )}
+        />
+        <Controller
+          name="total"
+          control={control}
+          render={({ field }) => (
+            <Component.TextField
+              id="total"
+              label="Total"
+              variant="outlined"
+              disabled
+              {...field}
+            />
+          )}
+        />
+      </div>
+      <div className={styles.buttonContainer}>
+        <Component.Button type="submit" variant="contained" color="primary">
```

Comment:
One UX required here. I shouldn't be able to click that button if for is not initialy changed, empty or not valid.